libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/guidance/junction/api/MapboxJunctionApiTest.kt

```diff
@@ -0,0 +1,258 @@
+package com.mapbox.navigation.ui.maps.guidance.junction.api
+
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.common.HttpRequest
+import com.mapbox.common.HttpRequestError
+import com.mapbox.common.HttpResponse
+import com.mapbox.common.HttpResponseCallback
+import com.mapbox.common.HttpResponseData
+import com.mapbox.common.HttpServiceInterface
+import com.mapbox.common.core.module.CommonSingletonModuleProvider
+import com.mapbox.navigation.testing.MainCoroutineRule
+import com.mapbox.navigation.ui.base.model.Expected
+import com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionAction
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionProcessor
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionResult
+import com.mapbox.navigation.ui.maps.guidance.junction.model.JunctionError
+import com.mapbox.navigation.ui.maps.guidance.junction.model.JunctionValue
+import com.mapbox.navigation.utils.internal.ThreadController
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.slot
+import io.mockk.unmockkObject
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@ExperimentalCoroutinesApi
+class MapboxJunctionApiTest {
+
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
+    private val consumer:
+        MapboxNavigationConsumer<Expected<JunctionValue, JunctionError>> = mockk(relaxed = true)
+    private val bannerInstructions: BannerInstructions = mockk()
+    private val junctionApi = MapboxJunctionApi("pk.1234")
+
+    @Before
+    fun setUp() {
+        mockkObject(ThreadController)
+        every { ThreadController.IODispatcher } returns coroutineRule.testDispatcher
+        mockkObject(JunctionProcessor)
+        mockkObject(CommonSingletonModuleProvider)
+    }
+
+    @After
+    fun tearDown() {
+        unmockkObject(JunctionProcessor)
+        unmockkObject(CommonSingletonModuleProvider)
+        unmockkObject(ThreadController)
+    }
+
+    @Test
+    fun `process state signboard unavailable`() {
+        every {
+            JunctionProcessor.process(
+                JunctionAction.CheckJunctionAvailability(bannerInstructions)
+            )
+        } returns JunctionResult.JunctionUnavailable
+        val messageSlot = slot<Expected.Failure<JunctionError>>()
+
+        junctionApi.generateJunction(bannerInstructions, consumer)
+
+        verify(exactly = 1) { consumer.accept(capture(messageSlot)) }
+        Assert.assertEquals(
```

Comment:
```suggestion
        assertEquals(
```

Same thing applies to the rest of the tests below 👇